apps/funke/use-cases/ReceivePidUseCase.ts

```diff
@@ -0,0 +1,181 @@
+import type { AppAgent } from '@/agent'
+import { AusweisAuthFlow } from '@animo-id/expo-ausweis-sdk'
+import {
+  type OpenId4VciRequestTokenResponse,
+  type OpenId4VciResolvedAuthorizationRequest,
+  type OpenId4VciResolvedCredentialOffer,
+  acquireAccessToken,
+  receiveCredentialFromOpenId4VciOffer,
+  resolveOpenId4VciOffer,
+} from '@package/agent'
+
+export interface ReceivePidUseCaseOptions {
+  agent: AppAgent
+  onStateChange?: (newState: ReceivePidUseCaseState) => void
+}
+
+export type ReceivePidUseCaseState = 'id-card-auth' | 'acquire-access-token' | 'retrieve-credential' | 'error'
+
+export class ReceivePidUseCase {
+  private agent: AppAgent
+
+  private resolvedCredentialOffer: OpenId4VciResolvedCredentialOffer
+  private resolvedAuthorizationRequest: OpenId4VciResolvedAuthorizationRequest
+  private idCardAuthFlow: AusweisAuthFlow
+  private accessToken?: OpenId4VciRequestTokenResponse
+
+  private onStateChange?: (newState: ReceivePidUseCaseState) => void
+  private currentState: ReceivePidUseCaseState = 'id-card-auth'
+  public get state() {
+    return this.currentState
+  }
+
+  private static SD_JWT_VC_OFFER =
```

Comment:
Are these static values permament or just for testing?

Reply:
permanent.